segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/stat/AutoScalerConfig.java

```diff
@@ -27,6 +27,12 @@
     public static final Property<Integer> CACHE_CLEANUP_IN_SECONDS = Property.named("cacheCleanUpInSeconds", 5 * 60);
     public static final Property<Integer> CACHE_EXPIRY_IN_SECONDS = Property.named("cacheExpiryInSeconds", 20 * 60);
     public static final Property<String> CONTROLLER_URI = Property.named("controllerUri", "tcp://localhost:9090");
+    public static final Property<Boolean> TLS_ENABLED = Property.named("tlsEnabled", false);
+    public static final Property<String> TLS_CERT_FILE = Property.named("tlsCertFile", "");
+    public static final Property<Boolean> AUTH_ENABLED = Property.named("authEnabled", false);
+    public static final Property<String> AUTH_USERNAME = Property.named("authUsername", "");
+    public static final Property<String> AUTH_PASSWORD = Property.named("authPassword", "");
+    public static final Property<String> TOKEN_SIGNING_KEY = Property.named("tokenSigningKey", "secret");
```

Comment:
authPassword and tokenSigningKey are passed as jvm options `-DautoScale.authPassword` `-DautoScale.tokenSigningKey` on the command line.    Could these become filenames to files that actually contain these secrets?  These way they are not exposed on  a `ps` command on the host?

Reply:
This makes sense. Will update the PR to take these parameters from a file.

Reply:
#2311 is an issue to track this. I do not want to add any more functionality/fixes to this PR. Will implement this in another branch.

Reply:
That makes sense.  I do agree the more we make changes/improvements in this PR, the less likely it is to ever make it to master. :)